keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandler.kt

```diff
@@ -132,7 +136,34 @@ class SecurityGroupHandler(
               response.accountName,
               response.region,
               response.vpcId?.let { cloudDriverCache.networkBy(it).name },
-              response.description
+              response.description,
+              response.inboundRules.flatMap { rule ->
+                val ingressGroup = rule.securityGroup
+                val ingressRange = rule.range
+                when {
+                  ingressGroup != null -> rule.portRanges?.map { portRange ->
+                    ReferenceSecurityGroupRule(
+                      Protocol.valueOf(rule.protocol!!.toUpperCase()),
+                      if (ingressGroup.name != response.name || ingressGroup.accountName != response.accountName) ingressGroup.name else null,
+                      if (ingressGroup.accountName != response.accountName) ingressGroup.accountName else null,
+                      if (ingressGroup.vpcId != response.vpcId) cloudDriverCache.networkBy(ingressGroup.vpcId!!).name else null,
```

Comment:
This is kinda nasty. Basically if the desired state is a self referencing or non-cross-account rule you can omit these fields. But if we include them in the converted response from River we'll get a false delta and keep no-op upserting the security group.

I think long term I'd like to define new sub-classes of `SecurityGroupRule` for the self-reference and in-account cases.